### PR TITLE
Fix and update Windows 10/11 22H2 date

### PR DESCRIPTION
**Pull request by TianQiBuTian**
### Description

Fix and update Windows 10/11 22H2 date
### Motivation and Context

1. https://docs.microsoft.com/windows/release-health/windows11-release-information

2. https://learn.microsoft.com/windows/release-health/release-information
3. https://docs.microsoft.com/lifecycle/announcements/windows-10-22h2-end-of-support-update

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: checks/utils/windowsversions.py [hunk 1]
```diff
@@ -40,7 +41,9 @@
 #
 # We probably don't need all the info here, but it comes in handy
 #
-# Ref: https://docs.microsoft.com/en-us/lifecycle/products/windows-10-home-and-pro
+# Ref: https://docs.microsoft.com/lifecycle/products/windows-10-home-and-pro
+#      https://aka.ms/WindowsUpdateHistory
+#      https://docs.microsoft.com/windows/release-health/release-information
```
Comment (RytoEX) [inline checks/utils/windowsversions.py:46]: If updating the docs, use the learn.microsoft.com domain instead (which is what these redirect to).

File: checks/utils/windowsversions.py [hunk 2]
```diff
@@ -17,20 +17,21 @@
...
+#      https://aka.ms/Windows11UpdateHistory
+#      https://docs.microsoft.com/windows/release-health/windows11-release-information
 win11versions = {
     22000: {
         "release": 2009,
         "name": "Windows 11 21H2",
         "date": datetime.date(2021, 10, 4),
         "EoS": datetime.date(2023, 10, 10),
     },
-    # Note: 22622 is 22H2 but with additional features coming 2023
```
Comment (RytoEX) [inline checks/utils/windowsversions.py:28]: Why remove this?

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: checks/utils/windowsversions.py [hunk 3]
```diff
@@ -123,8 +126,8 @@
     19045: {
         "release": 2009,
         "name": "Windows 10 22H2",
-        "date": datetime.date(2022, 10, 17),
-        "EoS": datetime.date(2024, 5, 14),
+        "date": datetime.date(2022, 9, 20),
```
[the remark this answers is not in the document]
Comment (WizardCM) [reply checks/utils/windowsversions.py:129]: Actually, the off-by-one issue is likely my fault due to ~ timezones ~.

[the remark this answers is not in the document]
Comment (TianQiBuTian) [reply checks/utils/windowsversions.py:129]: I don't know which one is actually correct. The 1703 version written by Microsoft also has two release dates, one is 2017/4/5 (support.microsoft.com) and the other is 2017/4/11 (learn.microsoft.com), while 'windowsversions.py' uses the former.

File: checks/utils/windowsversions.py (hunk 2 above)
Comment (TianQiBuTian) [reply checks/utils/windowsversions.py:28]: The current build number for Windows 11 22H2 beta channel is 22631. And there are also 22623 and 22624 between 22622 and 22631. Unless the beta channel is also added, this comment can be deleted.
https://blogs.windows.com/windows-insider/2023/05/25/announcing-windows-11-insider-preview-build-22621-1825-and-22631-1825/

Comment (RytoEX) [reply checks/utils/windowsversions.py:28]: The comment is just informative, as far as I know. 22622 (and up, for now) will also identify as "Windows 11" and might even show as "22H2" in some tools. The comment itself is not meant for us to add entries for Insider revisions/builds.

File: checks/utils/windowsversions.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RytoEX) [reply checks/utils/windowsversions.py:129]: There's really no need to fuss over past release dates. They are mostly informative, and mostly only important if the release date is in the future.

That said, I'd assume that support.microsoft.com provides earlier release dates than the GA dates listed everywhere else, perhaps due to provision that only Microsoft knows. That or those pages are simply incorrect, which would not be the first time.

**Review by RytoEX — APPROVED** [review]

**Pull request merged**